File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        HashMap<Integer, Integer> mons = new HashMap<>();
        
        for(int n : nums) {
            if (mons.containsKey(n)) {
                mons.put(n, mons.get(n) + 1);
            } else {
                mons.put(n, 1);
            }
        }
        
        List<Integer> valueList = new ArrayList<>(mons.values());
        Collections.sort(valueList);
        
        int num = nums.length / 2;
        int answer = 0;
        for(int v: valueList) {
            answer += 1;
            num -= 1;
            
            if (num <= 0) break;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    mon = {}
    
    for n in nums:
        if n in mon:
            mon[n] += 1
        else:
            mon[n] = 1
    
    num = len(nums) / 2
    val = sorted(mon.values())
    answer = 0
    
    print(val)
    for v in val:
        answer += 1
        num -= 1
        if num <= 0:
            break
            
    return answer